modules/core/src/main/java/org/apache/ignite/internal/processors/query/QueryTypeDescriptorImpl.java

```diff
@@ -582,12 +585,31 @@ else if (F.eq(prop.name(), valFieldName) || (valFieldName == null && F.eq(prop.n
 
             if (prop.precision() != -1 &&
                 propVal != null &&
-                String.class == propVal.getClass() && 
+                String.class == propVal.getClass() &&
                 ((String)propVal).length() > prop.precision()) {
-                throw new IgniteSQLException("Value for a column '" + prop.name() + "' is too long. " + 
+                throw new IgniteSQLException("Value for a column '" + prop.name() + "' is too long. " +
                     "Maximum length: " + prop.precision() + ", actual length: " + ((CharSequence)propVal).length(),
                     isKey ? TOO_LONG_KEY : TOO_LONG_VALUE);
             }
+
+            if (prop.precision() != -1 &&
+                propVal != null &&
+                BigDecimal.class == propVal.getClass() &&
+                ((BigDecimal)propVal).precision() > prop.precision()) {
+                throw new IgniteSQLException("Value for a column '" + prop.name() + "' is too long. " +
```

Comment:
Value for a column  -> Value for a colum XXX is out of range. Maximum precision: YYY, actual precision: ZZZ